src/librustc/middle/cstore.rs

```diff
@@ -191,6 +191,8 @@ pub trait MetadataLoader {
                           -> Result<MetadataRef, String>;
 }
 
+pub type MetadataLoaderDyn = dyn MetadataLoader + Sync;
```

Comment:
Do we need this to be Sync even in the non-parallel compiler case? Maybe we want `sync::Sync` here which is either the proper marker trait from core or the data structures one which is just auto for everything?

Reply:
The metadata loaders have no fields, so it doesn't matter which `Sync` trait is used.